### ISSUE TITLE
$translate.onReady resolves promise too early when used with angular-translate-storage-local

### ISSUE THREAD

**Issue by aliok** [angular-translate/angular-translate#1402]
I always get `Translate ready, current language : undefined` with the following code snippets.

When I use `$timeout` instead of `$translate.onReady()`, I see the language returned correctly by `$translate.use()`.

Here is the relevant part from my bower.json:

```json
"angular-translate": "~2.9.1",
"angular-translate-loader-static-files": "~2.9.1",
"angular-translate-storage-local": "~2.9.1",
```

`app.js`:

```javascript
    angular
        .module('app', ['pascalprecht.translate', '...'])
        .run(run);

    run.$inject = ['$translate'];
    function run($translate) {
        $translate.onReady().then(function(){
            $log.debug("Translate ready, current language : " +  $translate.use());
        });
    }
```

`app.config.js`

```javascript
    angular
        .module('app')
        .config(config);

    config.$inject = ['$translateProvider'];
    function config($translateProvider) {
        $translateProvider.useStaticFilesLoader({
            prefix: 'shared/i18n/',
            suffix: '.json'
        });
        $translateProvider.useSanitizeValueStrategy('escape');
        $translateProvider
            .preferredLanguage('en')
            .fallbackLanguage(['en']);
        $translateProvider.useLocalStorage();
    }
```

**Comment by knalli** [comment]
I cannot confirm this, I've made a working demo and don't see an issue: http://plnkr.co/edit/C7guipxtT6AXzJhVHEV1?p=preview

**Comment by knalli** [comment]
Closed due inactivity.

**Comment by geo242** [comment]
@aliok  I experienced this issue as well, but the real issue ended up being that my language aliases didn't match anything, so the determinePreferredLanguage function was returning undefined, which meant that the preferred language wasn't available on the first call to $translate.use(). Adding a timeout "fixed" it, but once I fixed my available language key aliases, it all worked without a timeout. Hope this helps.
